File: typescript/packages/bench/src/rules/slotOverbookingRule.ts
/** Cardinality fixture: nominal (class, `implements`) implementor of RuleContract. */

import type { RuleContract } from "../contracts/ruleContract";
import type { Dataset } from "../dataset";

export class SlotOverbookingRule implements RuleContract {
    readonly key = "slot-overbooking";

    evaluate(data: Dataset): boolean {
        return data.technicians.length >= 1;
    }
}

File: rust/crates/atelier-bench/src/rules/slot_overbooking_rule.rs
//! Rule 23/24 (textual half).

use crate::contracts::rule_contract::RuleContract;
use crate::dataset::Dataset;

pub struct SlotOverbookingRule;

impl RuleContract for SlotOverbookingRule {
    fn evaluate(&self, data: &Dataset) -> bool {
        !data.orders.is_empty()
    }
}
